### ISSUE TITLE
Why not set ContentResponse.Position back to 0 after sdk deserializes the response

### ISSUE THREAD

**Issue by maririos** [Azure/azure-sdk-for-net#17711]
Curious why after the SDK reads the `ContentResponse`, it doesn't set the `Position` back to 0 in case the user wants to get the raw response and read it.
I see there is a sample [here](
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Response.md#accessing-http-response-content).

Context: Question from one of our FR customers that was thinking that the SDK doesn't allow for them to access the raw response.
Will put the issue in the FR label for now just to keep track of it

**Comment by maririos** [comment]
@pakrym I'm sure there is a reason, could you comment here?

**Comment by pakrym** [comment]
> I'm sure there is a reason, could you comment here?

It's not something we spent time considering, honestly. It's not always possible to reset the stream (in cases where it's not buffered) so I guess we decided not to even try.

On a positive note, a new property is being added to Response in https://github.com/Azure/azure-sdk-for-net/pull/19417. The `Content` property would represent the entire buffered response and would not be susceptible to the rewinding issue.
